cardano-community/guild-operators
# merge logmonitor and startup logmonitor

Combined functionality of logMonitor and startupLogMonitor into a single script.

# Testing

## Requirements
* OS dependencies
* cardano binaries (built or from release)
* CNCLI

## Checkout the code

Use `guild-deploy.sh` to target the branch.
```
./guild-deploy.sh -b combine-log-monitoring
```
Use `deploy-as-systemd.sh` to setup at minimum cnode and logmonitor. 

## Testing original functionality

The **RestartSec** has been lowered to 1 second. Sourcing was set to offline mode with `env offline`. The change in timing allows tracking chunk validation and ledger replay events.

1. Start/Stop/Kill the cnode.service.
2. Verify cnode-logmonitor.service responds accordingly with 1 second Restart vs. 20.
3. Check blocklog db for expected blocks.

## Testing specific to startup log
1. Pre Initialize blocklogdb
   * `/opt/cardano/cnode/scripts/cncli.sh init`
2. Check blocklog db for replay and validation table entries.
   * If the cardano node was cleanly stopped there will be minimal or no entries in replay or validation tables upon startup. Use `kill -9` on the pid of your cardano-node to cause an unclean shutdown and the next time the node starts it will perform chunk validation and replay the ledger and the associated tables will have entries.
